incremental-font-transfer/src/glyph_keyed.rs

```diff
@@ -213,74 +231,74 @@ fn retained_glyphs_in_font(
         })
 }
 
-fn retained_glyphs_total_size(
+fn retained_glyphs_total_size<T: GlyphDataOffsetArray>(
     gids: &IntSet<GlyphId>,
-    loca: &Loca,
+    offsets: &T,
     max_glyph_id: GlyphId,
 ) -> Result<u64, PatchingError> {
     let mut total_size = 0u64;
     for keep_range in retained_glyphs_in_font(gids, max_glyph_id) {
-        let start = keep_range.start();
-        let end = keep_range.end();
+        let start = *keep_range.start();
+        let end: GlyphId = keep_range
+            .end()
+            .to_u32()
+            .checked_add(1)
+            .ok_or(PatchingError::InternalError)?
+            .into();
 
-        let start_offset = loca
-            .get_raw(start.to_u32() as usize)
-            .ok_or(PatchingError::InvalidPatch("Start loca entry is missing."))?;
-        let end_offset = loca
-            .get_raw(end.to_u32() as usize + 1)
-            .ok_or(PatchingError::InvalidPatch("End loca entry is missing."))?;
+        let start_offset = offsets.offset_for(start)?;
+        let end_offset = offsets.offset_for(end)?;
 
         total_size += end_offset
             .checked_sub(start_offset) // TODO: this can be removed if we pre-verify ascending order
             .ok_or(PatchingError::FontParsingFailed(ReadError::MalformedData(
-                "loca entries are not in ascending order",
+                "offset entries are not in ascending order",
             )))? as u64;
     }
 
     Ok(total_size)
 }
 
-trait LocaOffset {
+trait WritableOffset<const DIV: usize> {
```

Comment:
I think the DIV param is worth a comment, it feels slightly mysterious (where is it used, what are the expected values?)